### PR TITLE
[FEATURE] Github statistics ingestion for Golang.

### PR DESCRIPTION
**Pull request by jparsai**
# Description

Introduced new flow and tasks to ingest Github details into graph db.

## Ticket
APPAI-1485

## Type of change

- [x] Introduced a new flow.
- [x] Introduced new tasks under newly created flow.
- [x] Created a new storage class for storing result of github task into S3 bucket.
- [x] 100% test coverage for new modules.
- [x] Changed validate_url function from private to public as it's been reused in new NewInitPackageFlow task.

## Note:
Flow/Task/Class/File names are not given proper names yet. Proper naming convention will be followed while creating ingestion flow as suggested.
For Data Importer API new API is in progress, hence existing API is used for now, it works fine as we are only ingesting github details having existing schema.
To ingest latest version/description schema changes are required in worker, which can be done while creating ingestion flow along and all necessary details can be added in new schema.

### CODE REVIEW

File: f8a_worker/workers/graph_importer_new.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    """Ingest to graph node."""
+
+    def execute(self, arguments):
+        """Task code.
+
+        :param arguments: dictionary with task arguments
+        :return: {}, results
+        """
+        self._strict_assert(arguments.get('ecosystem'))
+        self._strict_assert(arguments.get('name'))
```
Comment (yzainee) [inline f8a_worker/workers/graph_importer_new.py:29]: check for github url is missing

**Review by yzainee — CHANGES_REQUESTED** [review]

File: f8a_worker/workers/graph_importer_new.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""Ingest to graph task."""
```
[the remark this answers is not in the document]
Comment (yzainee) [reply f8a_worker/workers/graph_importer_new.py:1]: will leave it to you if you want to rename or keep it like this

File: f8a_worker/workers/init_package_flow_new.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    """Initialize package-level analysis."""
+
+    def execute(self, arguments):
+        """Task code.
+
+        :param arguments: dictionary with task arguments
+        :return: {}, results
+        """
+        self._strict_assert(isinstance(arguments.get('ecosystem'), str))
+        self._strict_assert(isinstance(arguments.get('name'), str))
```
[the remark this answers is not in the document]
Comment (yzainee) [reply f8a_worker/workers/init_package_flow_new.py:21]: check for github url is missing

File: f8a_worker/dispatcher/nodes.yml [hunk 4]
```diff
@@ -188,6 +188,22 @@
       queue: '{DEPLOYMENT_PREFIX}_{WORKER_ADMINISTRATION_REGION}_ReportGenerationTask_v0'
       storage: 'BayesianPostgres'
 
+    - name: 'NewInitPackageFlow'
+      import: 'f8a_worker.workers'
+      max_retry: 0
+      queue: '{DEPLOYMENT_PREFIX}_{WORKER_ADMINISTRATION_REGION}_NewInitPackageFlow_v0'
+    - name: 'NewGithubDetails'
```
[the remark this answers is not in the document]
Comment (jparsai) [reply f8a_worker/dispatcher/nodes.yml:195]: Yes, 'newPackageFlow' will be called with list of selective sub tasks of the flow needs to be executed.  
`selinon.run_flow_selective('flow_name', [<list of sub tasks of given flow>], node_args,....)`
Will be creating an API in jobs to run selective tasks, github can call the that API

File: f8a_worker/workers/init_package_flow_new.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jparsai) [reply f8a_worker/workers/init_package_flow_new.py:21]: For package level analysis we may not have version, even currently we drop version in init_ackage_flow
```
        # get rid of version if scheduled from the core analyses
        arguments.pop('version', None)
        arguments.pop('document_id', None)

```
In we put condition will not be able to proceed.
In ingestion flow (init_analysis_flow) version is required there this condition is must.
Please give your comment, if you still want version to be checked.

File: f8a_worker/workers/graph_importer_new.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    def execute(self, arguments):
+        """Task code.
+
+        :param arguments: dictionary with task arguments
+        :return: {}, results
+        """
+        self._strict_assert(arguments.get('ecosystem'))
+        self._strict_assert(arguments.get('name'))
+
+        package_list = [
```
[the remark this answers is not in the document]
Comment (jparsai) [reply f8a_worker/workers/graph_importer_new.py:31]: For now in data importer existing API is used, we are only ingesting github details having existing schema so it works fine, As suggested description/latest version can be set along with github details, but for it schema changes are required which will take some time to understand, modify and test, once these modifications are done new data importer API can be used which will update description, version, github details etc at once.

File: f8a_worker/workers/init_package_flow_new.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yzainee) [reply f8a_worker/workers/init_package_flow_new.py:21]: ok. i got the point. when we start the ingestion part, we will need it

**Review by yzainee — APPROVED** [review]
Changes look fine. Please cross check if we need ingest_to_graph or selective_ingest from data importer

**Pull request merged**